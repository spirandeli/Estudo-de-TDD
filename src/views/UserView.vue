<script>
import VUserSearchFormVue from '@/components/VUserSearchForm.vue'
import VUserProfile from '@/components/VUserProfile.vue'
export default {
  components: { VUserProfile, VUserSearchFormVue },
  name: 'UserView'
}
</script>

<template>
  <div>
    Hello World
    <VUserSearchFormVue />
    <VUserProfile />
  </div>
</template>