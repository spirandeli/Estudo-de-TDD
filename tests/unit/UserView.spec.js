import VUserProfile from '@/components/VUserProfile'
import VUserSearchForm from '@/components/VUserSearchForm'
import UserView from '@/views/UserView'
import { shallowMount } from '@vue/test-utils'

describe('UserView', () => {
  it('Se o componente renderiza', () => {
    // arrange
    const wrapper = shallowMount(UserView)

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Se os componentes estÃ£o sendo renderizados', () => {
    // arrange
    const wrapper = shallowMount(UserView)
    const userSearchForm = wrapper.find(VUserSearchForm)
    const userProfile = wrapper.find(VUserProfile)

    // assert
    expect(userSearchForm.exists()).toBe(true)
    expect(userProfile.exists()).toBe(true)
  })
})