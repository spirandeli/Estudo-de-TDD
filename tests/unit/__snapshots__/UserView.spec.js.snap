// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView Se o componente renderiza 1`] = `
<div>
  Hello World
  <vusersearchformvue-stub></vusersearchformvue-stub>
  <vuserprofile-stub></vuserprofile-stub>
</div>
`;
